### ISSUE TITLE
Posenet

### ISSUE THREAD

**Issue by Vaportoxic1** [Lavreniuk/SPIdepth#23]
I would like to request the posenet weights if you don't mind

**Comment by Lavreniuk** [comment]
@Vaportoxic1 , please find it here (just uploaded)
https://huggingface.co/MykolaL/SPIdepth/tree/main/kitti

**Comment by Vaportoxic1** [comment]
Hello bro thank you for your timely replies

**Comment by Vaportoxic1** [comment]
Hello thank you for your timely reply 
I have this  question why is it hard to recreate cityscapes results? even trying to recreate paper you modified sqldepth cityscapes, the lowest I could get was abs_rel 0.132  and that was with convnext large

**Comment by Vaportoxic1** [comment]
I honestly can't seem to figure out how it was done I fine-tuned with kitti still no where near close , is there another fine-tuning after the first fine-tuning, should I set t("--disable_automasking",
                                 help="if set, doesn't do auto-masking",
                                 action="store_true")   to store_false? please help me

**Comment by Lavreniuk** [comment]
make sure you have started with already finetuned weights of encoder and decoder on KITTI, not from scratch?

**Comment by Vaportoxic1** [comment]
ooh wait so I should use the finetuned weights of KITTI, not the self supervised weights?  (basically the weights trained with improved kitti ground truths")?

**Comment by Vaportoxic1** [comment]
Hello, bro
please help me on this one am really stuck by fine-tuned weights of KITTI, do you mean the supervised KITTI weights. The reason am asking this, is because the way the weights for fine-tuned model are saved is different, it is saved as one unified model and so I want to be sure if I need to modify the code and save the code and the decoder separately,  thank you so much in advance,

**Comment by Lavreniuk** [comment]
So process should be: 1. self supervision on KITTI. 2. finetune on KITTI. 3. finetune on Citiscapes.
so for citiscapes you need to init weights from step 2 (if you need the weighs, please check https://huggingface.co/MykolaL/SPIdepth/tree/main/cityscapes), if you want weights from step 2. please check (https://huggingface.co/MykolaL/SPIdepth/tree/main/kitti)
If you want to train by yourself and what to convert weights, please check: [ckpt_to_pth.py](https://github.com/Lavreniuk/SPIdepth/blob/main/finetune/ckpt_to_pth.py)
hope this clarify the process and helps you.

**Comment by Wangzi06** [comment]
Bro you are a life save thank you so much

**Comment by Lavreniuk** [comment]
@Wangzi06 , I am glad that you found it helpful for you :)

**Comment by Wangzi06** [comment]
One last question, I train self supervised kitti with some changes on the encoder and decoder, the results were very close to the original results when I fine-tuned during fine-tuning it showed consisted performance increase very good slightly better than the original. but when I changed the weights from chkpt to pth and did the evaluation the results where holibly bad. why would this happen.

**Comment by Lavreniuk** [comment]
@Wangzi06, I am glad to hear that you could not only reproduce, but to got even better results!
To be honest I don't know, maybe it is somehow related to your modification? Also did you pick for converting the best epoch, not last one?

**Comment by Wangzi06** [comment]
![Image](https://github.com/user-attachments/assets/efdd9aeb-6edb-4fac-a2c6-cf68895fbe5b)  This the result when I evaluate your kitti model on make3d, I believe to problem might be how I did it not your model below is the code I used 

![Image](https://github.com/user-attachments/assets/04f7b6e4-22a5-420b-a67f-d787642840a7)

![Image](https://github.com/user-attachments/assets/00683c96-99ef-4098-ad17-58a5f832d8e2)

![Image](https://github.com/user-attachments/assets/f0de016b-c540-47fd-95a1-de8a85581f31)

![Image](https://github.com/user-attachments/assets/c4136c0f-e23f-4101-818d-2659e73003e6)

**Comment by Wangzi06** [comment]
hello @Lavreniuk  Please If you have spare time help figure where it went wrong. why can't I get the same results metrics as reported in the paper on make3D evaluation.
